projects/igniteui-angular/src/lib/services/transaction/igx-hierarchical-transaction.ts

```diff
@@ -65,10 +65,18 @@ export class IgxHierarchicalTransactionService<T extends HierarchicalTransaction
                 transactions = transactions.filter(t => t.id === id);
             }
             DataUtil.mergeHierarchicalTransactions(data, transactions, childDataKey, primaryKey, true);
+            this.clear(id);
         } else {
-            super.commit(data, id);
+            if (id !== undefined) {
+                super.commit(data, id);
+            } else {
+                this._states.forEach((s: S, key: any) => {
+                    if (this.updateRecord(data, s)) {
```

Comment:
IMO `updateRecord` will always return `true`. If I am right you can remove this check and make `updateRecord` to be `void` as it was.

Reply:
updateRecord will always return true only for flat grid. In hierarchical grid it will return false if the transaction to be applied is not applicable for the current child grid's data.